omni-relayer/src/workers/solana.rs

```diff
@@ -0,0 +1,326 @@
+use std::{str::FromStr, sync::Arc};
+
+use anyhow::Result;
+use futures::future::join_all;
+use log::{error, info, warn};
+
+use omni_connector::OmniConnector;
+use omni_types::{
+    locker_args::StorageDepositAction, prover_args::WormholeVerifyProofArgs,
+    prover_result::ProofKind, ChainKind, OmniAddress,
+};
+use solana_sdk::pubkey::Pubkey;
+
+use crate::{config, utils};
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct InitTransferWithTimestamp {
+    pub amount: u128,
+    pub token: String,
+    pub recipient: String,
```

Comment:
Not worth it. `InitTransferPayload` can't be parsed automatically, so I'd need to do it manually like I do it now, so I prefer to have all parsing logic inside workers instead of indexers